### PR TITLE
Fold SVGAnimatedPoints (a mixin) into SVGPolygonElement and SVGPolylineElement

### PR DESCRIPTION
**Pull request by foolip**
Redirect it to SVGPolygonElement, an arbitrary choice between the two.

Document animatedPoints in a similar way to animVal here:
https://developer.mozilla.org/en-US/docs/Web/API/SVGAnimatedString

This ignores that the spec requires animatedPoints to be an alias of 
points, because that does not match what's actually implemented (yet):
https://svgwg.org/svg2-draft/shapes.html#__svg__SVGAnimatedPoints__animatedPoints
https://wpt.fyi/results/svg/shapes/animatedPoints-non-animated.html?run_id=5709221590990848&run_id=5697986191425536&run_id=5747784122630144&run_id=5742635329257472

### CODE REVIEW

**Comment by foolip** [comment]
@Elchi3 I'm leaving this as a draft pull request since I unfortunately (per habit) looked too closely and discovered something terrible. `points` and `animatedPoints` are claimed to be aliases in the spec, but I checked Chromium's implementation and they're not aliases. What do do? 😢

**Comment by Elchi3** [comment]
> @Elchi3 I'm leaving this as a draft pull request since I unfortunately (per habit) looked too closely and discovered something terrible. `points` and `animatedPoints` are claimed to be aliases in the spec, but I checked Chromium's implementation and they're not aliases. What do do? 😢

Looks like a spec change per https://www.w3.org/TR/SVG2/shapes.html#InterfaceSVGAnimatedPoints

We probably want to show this spec change in the docs and in BCD then. Currently BCD just says "only implements the SVG 1.1 specification of the interface." which is too abstract. We should talk about this specific change instead I assume.

**Comment by foolip** [comment]
Aha, so this might be what the notes were about! I frowned and sent a PR to remove them in https://github.com/mdn/browser-compat-data/pull/10560, but maybe it should go together with demixing.

**Comment by foolip** [comment]
Judging by https://wpt.fyi/results/svg/shapes/animatedPoints-non-animated.html no browser has followed the spec change here, it's just something the spec editors hope for at this point. Should we just document what's actually implemented?

**Comment by Elchi3** [comment]
I see. Well, then I guess we should try to document reality and that's SVG 1.1 and implementations and not SVG 2.

**Comment by foolip** [comment]
@Elchi3 I've tried to document reality, PTAL.

File: files/en-us/web/api/svgpolygonelement/index.html [hunk 1]
```diff
@@ -17,7 +17,14 @@
 
 <h2 id="Properties">Properties</h2>
 
-<p><em>This interface doesn't implement any specific properties, but inherits properties from its parent, {{domxref("SVGGeometryElement")}} and also implements properties from {{domxref("SVGAnimatedPoints")}}.</em></p>
+<p><em>This interface also inherits properties from its parent, {{domxref("SVGGeometryElement")}}.</em></p>
+
+<dl>
+ <dt>{{domxref("SVGPolygonElement.animatedPoints")}} {{readOnlyInline}}</dt>
+ <dd>A {{DOMxRef("SVGPointList")}} representing the animated value of the element's {{SVGAttr("points")}} attribute. If the { {SVGAttr("points")}} attribute is not being animated, it contains the same value as the <code>points</code> property.</dd>
```
Comment (Elchi3) [inline files/en-us/web/api/svgpolygonelement/index.html:24]: I assume the build fails because you have a line break in a macro call.

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Thanks for doing this! There is a macro error but LGTM otherwise

File: files/en-us/web/api/svgpolygonelement/index.html (hunk 1 above)
Comment (foolip) [reply files/en-us/web/api/svgpolygonelement/index.html]: It took me a while to spot, but yeah I had `{ {`.

File: files/en-us/web/api/svgpolylineelement/index.html [hunk 2]
```diff
@@ -17,7 +17,14 @@
...
 <h2 id="Properties">Properties</h2>
 
-<p><em>This interface doesn't implement any specific properties, but inherits properties from its parent, {{domxref("SVGGeometryElement")}} and also implements properties from {{domxref("SVGAnimatedPoints")}}.</em></p>
+<p><em>This interface also inherits properties from its parent, {{domxref("SVGGeometryElement")}}.</em></p>
+
+<dl>
+  <dt>{{domxref("SVGPolylineElement.animatedPoints")}} {{readOnlyInline}}</dt>
+  <dd>A {{DOMxRef("SVGPointList")}} representing the animated value of the element's {{SVGAttr("points")}} attribute. If the {{SVGAttr("points")}} attribute is not being animated, it contains the same value as the <code>points</code> property.</dd>
+  <dt>{{domxref("SVGPolylineElement.points")}}</dt>
+  <dd>A {{DOMxRef("SVGPointList")}} representing the base (i.e., static) value of the element's {{SVGAttr("points")}} attribute. Modifications via the {{DOMxRef("SVGPointList")} object are reflected in the {{SVGAttr("points")}} attribute, and vice versa.</dd>
```
Comment (Elchi3) [inline files/en-us/web/api/svgpolylineelement/index.html:26]:
```suggestion
  <dd>A {{DOMxRef("SVGPointList")}} representing the base (i.e., static) value of the element's {{SVGAttr("points")}} attribute. Modifications via the {{DOMxRef("SVGPointList")}} object are reflected in the {{SVGAttr("points")}} attribute, and vice versa.</dd>
```

File: files/en-us/web/api/svgpolygonelement/index.html [hunk 3]
```diff
@@ -17,7 +17,14 @@
...
 <h2 id="Properties">Properties</h2>
 
-<p><em>This interface doesn't implement any specific properties, but inherits properties from its parent, {{domxref("SVGGeometryElement")}} and also implements properties from {{domxref("SVGAnimatedPoints")}}.</em></p>
+<p><em>This interface also inherits properties from its parent, {{domxref("SVGGeometryElement")}}.</em></p>
+
+<dl>
+ <dt>{{domxref("SVGPolygonElement.animatedPoints")}} {{readOnlyInline}}</dt>
+ <dd>A {{DOMxRef("SVGPointList")}} representing the animated value of the element's {{SVGAttr("points")}} attribute. If the {{SVGAttr("points")}} attribute is not being animated, it contains the same value as the <code>points</code> property.</dd>
+ <dt>{{domxref("SVGPolygonElement.points")}}</dt>
+ <dd>A {{DOMxRef("SVGPointList")}} representing the base (i.e., static) value of the element's {{SVGAttr("points")}} attribute. Modifications via the {{DOMxRef("SVGPointList")} object are reflected in the {{SVGAttr("points")}} attribute, and vice versa.</dd>
```
Comment (Elchi3) [inline files/en-us/web/api/svgpolygonelement/index.html:26]:
```suggestion
 <dd>A {{DOMxRef("SVGPointList")}} representing the base (i.e., static) value of the element's {{SVGAttr("points")}} attribute. Modifications via the {{DOMxRef("SVGPointList")}} object are reflected in the {{SVGAttr("points")}} attribute, and vice versa.</dd>
```

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Looking forward to the day we are in markdown and with as little kumascript macros as possible. :-)

**Comment by foolip** [comment]
> Looking forward to the day we are in markdown and with as little kumascript macros as possible. :-)

What will xrefs look like with Markdown? Won't it still be a macro that one can mess up?

**Comment by Elchi3** [comment]
> What will xrefs look like with Markdown? Won't it still be a macro that one can mess up?

Initially, yes, but I hope we won't have xref macros forever.

**Review by Elchi3 — APPROVED** [review]

**Pull request merged**